### ISSUE TITLE
[Enhancement] Add "Reload Integration" functionality

### ISSUE THREAD

**Issue by clau-bucur** [custom-components/ble_monitor#251]
I'm using YAML configuration and a "reload integration" feature (supported lately by most integrations) would be very helpful when changes are done. This would eliminate the need to restart HASS.

**Comment by Ernst79** [comment]
Thanks for reporting. I'll look into it. 

Note for myself. The reload function is available in YAML mode, when clicking on the three dots on the integration tile, but it doesn't reload changes made in the YAML file. It just reloads the old config. 
Reloading does work in UI (config flow) mode.

**Comment by Ernst79** [comment]
@clau-bucur Do you know which integrations allow you to reload your YAML configuration without a restart. You mentioned that this is supported by most integrations lately. If you know which one exactly, I can have a look how they do that.

**Comment by clau-bucur** [comment]
Yes, here is the command_line component: https://github.com/home-assistant/core/commit/215e3f2dab7611f369efe6de441a7aba745854ef

It's a pretty simple  change to be done.

**Comment by Ernst79** [comment]
Thanks, Strange that this isn't documented (or I'm looking a the wrong place).

http://dev-docs.home-assistant.io/en/master/api/helpers.html?highlight=reload#module-homeassistant.helpers.service

**Comment by Ernst79** [comment]
It's unfortunately not as simple as copying the example from command_line. Adding it under `load_platform` doesn't work, as we do not really use that part of the code (as we use config flow, unlike command_line), If I add the service in __init__.py or at setup_entry, it won't even start HA anymore when I add this. 

I'll have to look further to see if this is even possible.

**Comment by clau-bucur** [comment]
Lovely. Thanks for investigating. If it's not possible, it's not really that big of a deal. It's not likely that the integration needs to be reloaded too often anyway.

**Comment by Ernst79** [comment]
I decided to not fix this. It's too complicated to change and the use of YAML is becoming less frequently used. Sorry.
